packages/target-electron/static/webxdc-preload.js

```diff
@@ -291,17 +298,29 @@ class RealtimeListener {
           ipcRenderer.invoke('webxdc.exit')
         } catch (error) {
           loadingDiv.remove()
-          iframe.src = 'index.html'
+          iframe.src = locationUrl !== '' ? locationUrl : 'index.html'
           iframe.contentWindow.window.addEventListener(
             'keydown',
             keydown_handler
           )
+          connectionsFilled = true
         }
       } catch (error) {
         console.log('error loading, should crash/close window', error)
         ipcRenderer.invoke('webxdc.exit')
       }
     },
+    /**
+     * called via webContents.executeJavaScript
+     */
+    setLocationUrl(href) {
+      locationUrl = Buffer.from(href, 'base64').toString('utf8')
+      if (locationUrl && locationUrl !== '' && connectionsFilled) {
+        // if connectionsFilled is false, the url is loaded after
+        // the connections are filled
+        window.frames[0].window.location = locationUrl
+      }
+    },
```

Comment:
As said above. setLocationUrl does nothing that the app could not do directly.

But yes, the function could also run in the wrapper. I will have a look

Reply:
> setLocationUrl does nothing that the app could not do directly.

Yes, right now it does not look like it does, but it unnecessarily lives in a dangerous area with access to Node functions and variables defined above.

Reply:
As was discussed in private, implementing my suggestion requires rethinking the design, because of the fill 500 thing.
I can live with the current one, as long as we're careful with future changes.